### PR TITLE
Added breeding support

### PR DESCRIPTION
**Pull request by Fusezion**
### Description

This PR adds the basis of breeding into skript
```applescript
# Event
[on] [entity] breed[ing]

# Expressions
bred child
breeding mother
breeding father
```
image
*ignore the \<none\> was making sure `event-entity` = child*
`event-entity` acts the same as `bred child`
`event-livingentity` returns the entity who started the breeding
`event-item` returns the itemstack used for breeding

### Note
as for the experience changes I've copied the ones from the PR #4284 

---
**Target Minecraft Versions:**  1.12.2+ (Skript 2.7 will remove this requirement)
**Requirements:**  MC 1.12.2
**Related Issues:**  #5111

### CODE REVIEW

**Comment by Fusezion** [comment]
I'll see about adding the stuff under `breedable` later once I get a chance just burnt out on seeing the same stuff non stop

**Comment by TheLimeGlass** [comment]
There is already a pull request standardizing all the the event values to be enum.

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 1]
```diff
@@ -727,13 +728,13 @@ public Entity get(final PlayerInteractEntityEvent e) {
 			@Override
 			@Nullable
 			public ItemStack get(final PlayerInteractEntityEvent e) {
-        EquipmentSlot hand = e.getHand();
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:730]: If possible, can you revert these unrelated changes?

**Review by APickledWalrus — CHANGES_REQUESTED** [review]
Nice feature :D

I think ExprBredChild, ExprBreedingFather, and ExprBreedingMother should be combined into one expression (ex: ExprBreedingFamily) (idk a good name lol)

Once that expression is made, I'll do further review - please feel free to re-request my review

File: src/main/java/ch/njol/skript/expressions/ExprExperience.java [hunk 2]
```diff
@@ -124,13 +129,15 @@ else if (e instanceof BlockBreakEvent)
...
 			}
 		else
-			d = 0;
-		
-		d = Math.max(0, Math.round(d));
-		if (e instanceof ExperienceSpawnEvent)
-			((ExperienceSpawnEvent) e).setSpawnedXP((int) d);
+			experience = 0;
+
+		experience = Math.max(0, Math.round(experience));
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/expressions/ExprExperience.java:134]: You can't continue to have this round method. The round method originally existed for the double, which is now an integer from the looks of it. Math.round(integer) doesn't exist, it only does double and floats, so no clue how this is valid code unless it's somehow converting it to a double/float.

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]
Order of methods is `init -> get/getAll -> acceptChange -> change -> setTime -> getTime -> isSingle -> getReturnType -> toString` for SimpleExpression

The usage of event-father and event-mother can be used after Pickle's pull request at https://github.com/SkriptLang/Skript/pull/4963 to distinguish between the many entities that can be involved in this event as there are many conflicts.

**Comment by Fusezion** [comment]
Whoops , oh well

**Pull request closed without merging**